src/test/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaMapperTest.java

```diff
@@ -42,28 +48,56 @@ public void testDefaultTopic() {
     public void testSingleLevel() {
         List<MappingRule> rules = new ArrayList<>();
 
-        rules.add(new MappingRule("sensors/+/data", "sensor_data"));
-        rules.add(new MappingRule("devices/{device}/data", "devices_{device}_data"));
-        rules.add(new MappingRule("fleet/{fleet}/vehicle/{vehicle}", "fleet_{fleet}"));
-        rules.add(new MappingRule("building/{building}/floor/{floor}", "building.{building}.floor.{floor}"));
-        rules.add(new MappingRule("term/{number}", "term{number}"));
+        rules.add(new MappingRule("sensors/[^/]+/data", "sensor_data", null));
+        rules.add(new MappingRule("devices/([^/]+)/data", "devices_$1_data", "device_$1"));
+        rules.add(new MappingRule("fleet/([0-9]+)/vehicle/(\\w+)", "fleet_$1", "fleet$2"));
+        rules.add(new MappingRule("building/(\\d+)/floor/(\\d+).*", "building_$2_$1", "building_$1"));
```

Comment:
can we make this example test clearer, just using the placeholders in the "logical" order $1_$2? 
Then you can have another test case where you have a different order, to show that it's possible.